packages-ts/gauntlet-terra-contracts/src/commands/contracts/ocr2/proposal/acceptProposal.ts

```diff
@@ -1,25 +1,103 @@
 import { Result } from '@chainlink/gauntlet-core'
-import { logger } from '@chainlink/gauntlet-core/dist/utils'
-import { TransactionResponse } from '@chainlink/gauntlet-terra'
+import { logger, prompt } from '@chainlink/gauntlet-core/dist/utils'
+import { TransactionResponse, providerUtils, RDD } from '@chainlink/gauntlet-terra'
 import { CATEGORIES } from '../../../../lib/constants'
-import { AbstractInstruction, instructionToCommand } from '../../../abstract/executionWrapper'
+import { AbstractInstruction, instructionToCommand, BeforeExecute } from '../../../abstract/executionWrapper'
+import { printDiff } from '../../../../lib/diff'
+import { serializeOffchainConfig, deserializeConfig, generateSecretEncryptions } from '../../../../lib/encoding'
+import { getOffchainConfigInput, OffchainConfig } from '../proposeOffchainConfig'
+import { getLatestOCRConfig } from '../../../../lib/inspection'
+import Long from 'long'
+import assert from 'assert'
 
 type CommandInput = {
   proposalId: string
   digest: string
+  offchainConfig: OffchainConfig
 }
 
 type ContractInput = {
   id: string
   digest: string
 }
 
+const translateConfig = (rawOffchainConfig: any, additionalConfig?: any): OffchainConfig => {
```

Comment:
Here I tried to 'translate' the offchain config into comparable form, i.e. we can compare longs structs but it's not handy at all. Raw config is the `offchain_config` that was serialized and encoded, additional one is just additional parameters that we might want to use for comparison.

Reply:
Ok, then split the function in 2:
- `longsToNumber` should be an independent function. Be careful to make not mutate the input directly
```js
type LongObj = {
  [key: string]: Long | LongObj
}

type NumberObj = {
  [key: string]: number | NumberObj
}

const longsToNumber = (obj: LongMap): NumberObj => {
  return Object.entries(obj).reduce((agg, [key, value]) => {
    if (Long.isLong(value)) return { ...agg, ...{ [key]: (value as Long).toNumber() } }
    if (typeof value === 'object') return { ...agg,  ...longsToNumber(value) }
    return agg
  }, {} as NumberObj)
}
```

- Adding more properties to OffchainConfig should be limited to some properties. I could see `f` and `offchainPublicKeys` only. So make an function that only include those

Reply:
I didn't understand your suggestion here tbh. 
I have a non-homogeneous object (plain values, arrays, buffers, other objects) with potentially nested objects that can potentially be Longs. I don't need to make it plain with reduce, I just wanted to find all longs and convert them into readable numbers. It's safe from mutations either since `...` makes a new copy